backend/src/services/folder-commit/folder-commit-types.ts

```diff
@@ -0,0 +1,74 @@
+import { z } from "zod";
+
+const baseChangeSchema = z.object({
+  id: z.string(),
+  folderCommitId: z.string(),
+  changeType: z.string(),
+  isUpdate: z.boolean().optional(),
+  createdAt: z.union([z.string(), z.date()]),
+  updatedAt: z.union([z.string(), z.date()]),
+  actorMetadata: z
+    .union([
+      z.object({
+        id: z.string().optional(),
+        name: z.string().optional()
+      }),
+      z.unknown()
+    ])
+    .optional(),
+  actorType: z.string().optional(),
+  message: z.string().optional().nullable(),
+  folderId: z.string().optional()
+});
+
+const commitChangeSchema = baseChangeSchema.extend({
+  secretVersionId: z.string().optional().nullable(),
+  folderVersionId: z.string().optional().nullable(),
+  folderName: z.string().optional().nullable(),
+  folderChangeId: z.string().optional().nullable(),
+  secretKey: z.string().optional().nullable(),
+  secretVersion: z.union([z.string(), z.number()]).optional().nullable(),
+  secretId: z.string().optional().nullable(),
+  folderVersion: z.union([z.string(), z.number()]).optional().nullable(),
+  versions: z
+    .array(
+      z.object({
+        secretKey: z.string().optional().nullable(),
+        secretComment: z.string().optional().nullable(),
+        skipMultilineEncoding: z.boolean().optional().nullable(),
+        secretReminderRepeatDays: z.number().optional().nullable(),
+        secretReminderNote: z.string().optional().nullable(),
+        metadata: z.unknown().optional().nullable(),
+        tags: z.array(z.string()).optional().nullable(),
+        secretReminderRecipients: z.array(z.any()).optional().nullable(),
+        secretValue: z.string().optional().nullable(),
+        name: z.string().optional().nullable()
+      })
+    )
+    .optional()
+});
```

Comment:
we're still using a unified object here which is not ideal because its just a bunch of nullable fields. 

I was thinking something more like the following:
<img width="933" alt="image" src="https://github.com/user-attachments/assets/317afc22-085a-4c8f-8723-ccbb7eaa67ce" />